src/qibolab/instruments/rfsoc.py

```diff
@@ -1,54 +1,156 @@
-    Playing pulses requires first the execution of the ``setup`` function.
+HZ_TO_MHZ = 1e-6
+NS_TO_US = 1e-3
+
+
+def convert_qubit(qubit: Qubit) -> rfsoc.Qubit:
+    """Convert `qibolab.platforms.abstract.Qubit` to `qibosoq.abstract.Qubit`"""
+    if qubit.flux:
+        return rfsoc.Qubit(qubit.flux.bias, qubit.flux.ports[0][1])
+    return rfsoc.Qubit(0.0, None)
+
+
+def convert_pulse(pulse: Pulse, qubits: Dict) -> rfsoc.Pulse:
+    """Convert `qibolab.pulses.pulse` to `qibosoq.abstract.Pulse`"""
+
+    pulse_type = pulse.type.name.lower()
+    dac = getattr(qubits[pulse.qubit], pulse_type).ports[0][1]
+    adc = qubits[pulse.qubit].feedback.ports[0][1] if pulse_type == "readout" else None
+
+    try:
+        lo_frequency = getattr(qubits[pulse.qubit], pulse_type).local_oscillator._frequency
+    except NotImplementedError:
+        lo_frequency = 0
+
+    rfsoc_pulse = rfsoc.Pulse(
+        frequency=(pulse.frequency - lo_frequency) * HZ_TO_MHZ,
+        amplitude=pulse.amplitude,
+        relative_phase=np.degrees(pulse.relative_phase),
+        start=pulse.start * NS_TO_US,
+        duration=pulse.duration * NS_TO_US,
+        dac=dac,
+        adc=adc,
+        shape=None,
+        name=pulse.serial,
+        type=pulse_type,
+    )
+    if isinstance(pulse.shape, Rectangular):
+        rfsoc_pulse.shape = "rectangular"
+    elif isinstance(pulse.shape, Gaussian):
+        rfsoc_pulse.shape = "gaussian"
+        rfsoc_pulse.rel_sigma = pulse.shape.rel_sigma
+    elif isinstance(pulse.shape, Drag):
+        rfsoc_pulse.shape = "drag"
+        rfsoc_pulse.rel_sigma = pulse.shape.rel_sigma
+        rfsoc_pulse.beta = pulse.shape.beta
+    return rfsoc_pulse
+
+
+def convert_frequency_sweeper(sweeper: rfsoc.Sweeper, sequence: PulseSequence, qubits: Dict[int, Qubit]):
+    """Converts frequencies for `qibosoq.abstract.Sweeper` considering LO and HZ_TO_MHZ"""
+    for idx, jdx in enumerate(sweeper.indexes):
+        if sweeper.parameter[idx] is rfsoc.Parameter.frequency:
+            pulse = sequence[jdx]
+            try:
+                lo_frequency = getattr(qubits[pulse.qubit], pulse.type.name.lower()).local_oscillator._frequency
+            except NotImplementedError:
+                lo_frequency = 0
+
+            sweeper.starts[idx] = (sweeper.starts[idx] - lo_frequency) * HZ_TO_MHZ
+            sweeper.stops[idx] = (sweeper.stops[idx] - lo_frequency) * HZ_TO_MHZ
+
+
+def convert_sweep(sweeper: Sweeper, sequence: PulseSequence, qubits: List[Qubit]) -> rfsoc.Sweeper:
+    """Convert `qibolab.sweeper.Sweeper` to `qibosoq.abstract.Sweeper`"""
+
+    parameters = []
+    starts = []
+    stops = []
+    indexes = []
+
+    if sweeper.parameter is Parameter.bias:
+        for qubit in sweeper.qubits:
+            parameters.append(rfsoc.Parameter.bias)
+            for idx, seq_qubit in enumerate(qubits):
+                if qubit == seq_qubit:
+                    indexes.append(idx)
```

Comment:
```suggestion
            indexes.append(qubits.index(qubit))
```
This should be enough :)
https://docs.python.org/3/library/stdtypes.html#common-sequence-operations

Reply:
actually, qubits was a dictionary (there was indeed a bug). Not sure if there is a better way, now I did:
```python
indexes.append(list(qubits.values().index(qubit)))
```

Reply:
For the time being, it is fine like this (with the correct parenthesis written in the other comment).
If needed, we will improve later on.